src/passport.js

```diff
@@ -0,0 +1,68 @@
+const passwordGenerator = require('generate-password');
+const bcrypt = require('bcrypt');
+const { sendEmail } = require('./services/mail');
+const User = require('./models/user');
+
+const saltRounds = 10;
+
+const googleConfigs = {
+  clientID: process.env.GAPP_CLIENT_ID,
+  clientSecret: process.env.GAPP_CLIENT_SECRET,
+  callbackURL: 'http://localhost:3000/api/auth/google/callback',
+};
+
+const afterGoogleLogin = async function (
+  accessToken,
+  refreshToken,
+  profile,
+  cb
+) {
+  try {
+    let user = await User.findOne({ email: profile._json.email });
+    if (!user) {
+      const username = `${
+        profile._json.email.split('@')[0]
+      }_${passwordGenerator.generate({
+        length: 3,
+        numbers: true,
+        lowercase: false,
+        uppercase: false,
+      })}`;
+
+      const password = passwordGenerator.generate({
+        length: 10,
+        numbers: true,
+      });
+      const passwordHash = await bcrypt.hash(password, saltRounds);
+
+      user = await User.create({
+        username,
+        email: profile._json.email,
+        firstName: profile._json.given_name,
+        lastName: profile._json.family_name,
+        address: 'address',
+        password_hash: passwordHash,
+      });
+      const emailOptions = {
+        from: process.env.EMAIL_USER,
+        to: profile._json.email,
+        subject: 'Paying it forward password',
+        html: ` <h1>Hi ${profile._json.given_name},</h1>
+  <p>Here is your automatically generated password: ${password}</p>
```

Comment:
Here I think if you surround the password with a bold tag `<b>${password}</b>` it would be better to notice in the email. Also, I think you can send the user his username also